cmd/proxy/tcp_proxy.go

```diff
@@ -48,7 +48,8 @@ func (proxy *TCPProxy) clientLoop(client *net.TCPConn, quit chan bool) {
 				from.CloseWrite()
 			}
```

Comment:
````
 			if err, ok := err.(*net.OpError); ok && err.Err == syscall.EPIPE {
  				from.CloseWrite()
  			}
````
This looks unnecessary. But its not from this PR and can be addressed separately.

LGTM.